src/shared/azureagent/customroles.go

```diff
@@ -79,13 +81,20 @@ func (a *Agent) UpdateCustomRole(ctx context.Context, scope string, role *armaut
 		return to.Ptr(string(action))
 	})
 
+	// Compare the actions and dataActions to the existing role definition
```

Comment:
consider using some order-independent matching (to handle the case of equal-but-in-different-order).